h5io/_h5io.py

```diff
@@ -193,16 +188,22 @@ def _triage_write(key, value, root, comp_kw, where,
                       where + '.csr_matrix_shape', cleanup_data=cleanup_data,
                       slash=slash)
     else:
-        if isinstance(value, (DataFrame, Series)):
-            if isinstance(value, DataFrame):
-                title = 'pd_dataframe'
-            else:
-                title = 'pd_series'
-            rootname = root.name
-            cleanup_data.append({title: (rootname, key, value)})
-        else:
+        try:
+            from pandas import DataFrame, Series
+        except ImportError:
             err_str = 'unsupported type %s (in %s)' % (type(value), where)
             raise TypeError(err_str)
+        else:
+            if isinstance(value, (DataFrame, Series)):
+                if isinstance(value, DataFrame):
+                    title = 'pd_dataframe'
+                else:
+                    title = 'pd_series'
+                rootname = root.name
+                cleanup_data.append({title: (rootname, key, value)})
+            else:
+                err_str = 'unsupported type %s (in %s)' % (type(value), where)
+                raise TypeError(err_str)
```

Comment:
> Unindenting those lines doesn't quite work though because then we'd need to import pandas at the top of the entire if block.

Nah you can do it even with the current indentation level. Unfortunately I can't link to your previous commit (push forced?), but it had `return` at the end of the `if` clause. I think it was:
```
    else:
        try:
            from pandas import DataFrame, Series
        except ImportError:
            pass
        else:
            if isinstance(value, (DataFrame, Series)):
                if isinstance(value, DataFrame):
                    title = 'pd_dataframe'
                else:
                    title = 'pd_series'
                rootname = root.name
                cleanup_data.append({title: (rootname, key, value)})
                return
            err_str = 'unsupported type %s (in %s)' % (type(value), where)
            raise TypeError(err_str)
```
The problem with it was that in the `else` case, you don't `raise` but you should. Your solution here does raise, but it's not as DRY as what you could have gotten with this (what I meant to express by my un-indenting comment), which should be equivalent to what you have now, but with less code duplication:
```
    else:
        try:
            from pandas import DataFrame, Series
        except ImportError:
            pass
        else:
            if isinstance(value, (DataFrame, Series)):
                if isinstance(value, DataFrame):
                    title = 'pd_dataframe'
                else:
                    title = 'pd_series'
                rootname = root.name
                cleanup_data.append({title: (rootname, key, value)})
                return
        raise TypeError('unsupported type %s (in %s)' % (type(value), where))
```
See what I mean?

I think it's probably fine either way, so go with whichever you find more readable